onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc

```diff
@@ -119,9 +119,15 @@ Status ResizeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   std::vector<int32_t> sizes;
   std::vector<float> scales_hw;
   std::vector<int32_t> sizes_hw;
+  std::vector<int32_t> axes;
+  const bool isNhwc = model_builder.GetPreferredLayout() == DataLayout::NHWC;
   if (input_defs.size() == 3) {  // Use scales.
     ORT_RETURN_IF_NOT(GetResizeScales(initializers, node, scales, logger), "Error getting resize scales");
-    scales_hw = {scales[2], scales[3]};
```

Comment:
Do we check anywhere (I'm not seeing it) for other non-1 scales? ONNX has no notion of semantic layout for the resampling operator (every axis may have a valid scale, and no dimension means anything special), and so scales of [2,1,1,2] and [3,2] and [4,3,2] and even [5,1,1,1,4] are valid, but a scales of [4,3,2] is incompatible with WebNN which is limited to only 2D dimensions. Ideally we should just reshape an ONNX 2D or 3D tensor (given only two of the scales are non-1) to a WebNN 4D tensor with leading 1's, since they are all functionality equivalent. There are ONNX models with 2D/3D tensors. (it's fine to do this in a separate CR)

Reply:
@fdwr, I will do it in a separate PR.

Reply:
Okie dokie.